### DISCUSSION TITLE
A new edition?

### DISCUSSION THREAD

**Discussion by Jameel9** [rasbt/machine-learning-book#196] [General]
Hi Sebastian,

Thank you for your time and effort to produce and keep helping users/reader of this nice book. Do we expect a new edition that updates the contents and cover new emerging hot topics (such as MPNNs, diffusion models)?

**Comment by rasbt** [comment]
Thanks for asking and your interest in that! Right now, there is no new edition planned but hopefully some time in the upcoming years! The tricky part was that no more pages could be added to the current edition because the printer cannot handle it. But maybe swapping the GAN chapter by diffusion models etc. would be a good idea.

**Reply by RoyiAvital** [reply]
It would be great to have transformers in other context such as vision and time series.

**Reply by rasbt** [reply]
Agreed. It's just a bit tricky to decide which other content to remove so that the new content can be added. Thanks for suggesting!

**Reply by RoyiAvital** [reply]
Maybe the RL chapter? RL is totally different, better have its own book :-).

**Reply by rasbt** [reply]
Right, but it's one of the 3 broad categories of ML: unsupervised, supervised, and RL. Earlier editions (Python Machine Learning, Python Machine Learning 2nd edition) didn't have it, and readers were complaining about the lack of it 😆

**Reply by RoyiAvital** [reply]
Indeed. You know your audience the best.
If they had a choice, between the 2, which would they chose?

I assume that if you had a dedicated RL book (hint :-)) it would have made more sense.

**Reply by tc-git-1** [reply]
First, thanks for all you do to help so many people learn!

RE: "no more pages could be added to the current edition because the printer cannot handle it" 
and "It's ... tricky to decide which ... content to remove so that the new content can be added" 

Here are a few ideas to address the problem of page limits:
- Create an integrated series of two or more books about Machine Learning
- Create "online only" chapters
- Use things other than a traditional printed book to teach Machine Learning, such as one or more online self-paced courses on a platform such as Boot dot dev

**Reply by rasbt** [reply]
These are valid points:

> Create an integrated series of two or more books about Machine Learning

Yes, that's why I wrote a book just about LLMs, and now reasoning model techniques

> Create "online only" chapters

That's basically my substack 

> Use things other than a traditional printed book to teach Machine Learning, such as one or more online self-paced courses on a platform such as Boot dot dev

My ML and DL courses currently do this already

They are all legit points of course. But they don't solve the issue of the book page limit itself.
